gui/dump_all_dialogs.cpp

```diff
@@ -0,0 +1,85 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gui/dump_all_dialogs.h"
+
+#include "common/file.h"
+#include "common/language.h"
+#include "common/translation.h"
+
+#include "gui/ThemeEngine.h"
+#include "gui/gui-manager.h"
+#include "gui/launcher.h"
+
+#include "image/png.h"
+
+namespace GUI {
+
+void GUI::dumpDialog(const Common::String &dialogName, int res, const Common::String &lang) {
+	// Update GUI language
+	TransMan.setLanguage(lang);
+
+	// Update resolution
+	ConfMan.setInt("last_window_width", res, Common::ConfigManager::kApplicationDomain);
+	ConfMan.setInt("last_window_height", 600, Common::ConfigManager::kApplicationDomain);
+	g_system->initSize(res, 600);
+
+	// Create surface for displaying dialog
+	Graphics::Surface surf;
+	surf.create(g_system->getOverlayWidth(), g_system->getOverlayHeight(), g_system->getOverlayFormat());
+
+	GUI::LauncherChooser chooser;
+	chooser.selectLauncher();
+	chooser.runModal();
+	g_system->grabOverlay(surf);
```

Comment:
Starting from here, please put it into a separate method something like `saveGUISnapshot()`.